### ISSUE TITLE
Timeout on acquiring shared lock against resource of class

### ISSUE THREAD

**Issue by odbuser2** [orientechnologies/orientdb#2485]
OrientDB 1.7.3
Various OS but this is occurring on a Solaris 10 vm with storage that may have been briefly interrupted.

Once a timeout occurs on acquiring a shared lock, subsequent commits on that data fail with a rollback exception.  This application is multi-threaded and works in several environments.  In one of the environments, I think the file system is temporarily disrupted which causes the timeout in OrientDB.  Subsequent commits against the same data fail.  Other commits continue to work fine.  Although I do need to determine the cause of the possible temporary disruption, OrientDB should recover in this scenario.

Workaround:
Restart the application.

Below are three stack traces.  The first two happen around the time of the disruption and are two separate unrelated operations.  Subsequent attempts to commit to the same data fail and that is shown in the third section.  The third section occurs repeatedly on a schedule while the first two sections do not occur after the initial disruption.  Lots of other threads against different data (but same code) continue to work successfully.  It's only the ones that failed to acquire the lock that have the issue until the application is restarted.

Timeout on acquiring shared lock

```
Error on retrieving record #24:2 (cluster: clusterA)
com.orientechnologies.common.concur.OTimeoutException: Timeout on acquiring shared lock against resource of class : class com.orientechnologies.common.concur.resource.OSharedResourceAdaptiveExternal with timeout=30000
at com.orientechnologies.common.concur.resource.OSharedResourceAdaptive.acquireSharedLock(OSharedResourceAdaptive.java:152) ~[na:na]
at com.orientechnologies.common.concur.resource.OSharedResourceAdaptiveExternal.acquireSharedLock(OSharedResourceAdaptiveExternal.java:40) ~[na:na]
at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.readRecord(OLocalPaginatedStorage.java:1487) ~[na:na]
at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.readRecord(OLocalPaginatedStorage.java:768) ~[na:na]
at com.orientechnologies.orient.core.db.raw.ODatabaseRaw.read(ODatabaseRaw.java:246) ~[na:na]
... 17 common frames omitted
Wrapped by: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #24:2 (cluster: clusterA)
at com.orientechnologies.orient.core.db.raw.ODatabaseRaw.read(ODatabaseRaw.java:253) ~[na:na]
at com.orientechnologies.orient.core.db.record.ODatabaseRecordAbstract.executeReadRecord(ODatabaseRecordAbstract.java:1015) ~[na:na]
at com.orientechnologies.orient.core.tx.OTransactionOptimistic.loadRecord(OTransactionOptimistic.java:342) ~[na:na]
at com.orientechnologies.orient.core.db.record.ODatabaseRecordTx.load(ODatabaseRecordTx.java:248) ~[na:na]
at com.orientechnologies.orient.core.db.record.ODatabaseRecordTx.load(ODatabaseRecordTx.java:40) ~[na:na]
at com.orientechnologies.orient.core.record.impl.ODocument.field(ODocument.java:591) ~[na:na]
at com.tinkerpop.blueprints.impls.orient.OrientVertex.getVertices(OrientVertex.java:477) ~[na:na]
at com.tinkerpop.frames.annotations.AdjacencyAnnotationHandler.processVertex(AdjacencyAnnotationHandler.java:34) ~[na:na]
at com.tinkerpop.frames.annotations.AdjacencyAnnotationHandler.processElement(AdjacencyAnnotationHandler.java:26) ~[na:na]
at com.tinkerpop.frames.annotations.AdjacencyAnnotationHandler.processElement(AdjacencyAnnotationHandler.java:15) ~[na:na]
at com.tinkerpop.frames.FramedElement.invoke(FramedElement.java:89) ~[na:na]
... custom code...
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0]
at java.lang.Thread.run(Thread.java:744) [na:1.8.0]
```

Timeout on acquiring shared lock
Same error, different cluster approximately the same time as the other errors

```
Error on retrieving record #22:24 (cluster: clusterB)
com.orientechnologies.common.concur.OTimeoutException: Timeout on acquiring shared lock against resource of class : class com.orientechnologies.common.concur.resource.OSharedResourceAdaptiveExternal with timeout=30000
at com.orientechnologies.common.concur.resource.OSharedResourceAdaptive.acquireSharedLock(OSharedResourceAdaptive.java:152) ~[na:na]
at com.orientechnologies.common.concur.resource.OSharedResourceAdaptiveExternal.acquireSharedLock(OSharedResourceAdaptiveExternal.java:40) ~[na:na]
at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.readRecord(OLocalPaginatedStorage.java:1487) ~[na:na]
at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.readRecord(OLocalPaginatedStorage.java:768) ~[na:na]
at com.orientechnologies.orient.core.db.raw.ODatabaseRaw.read(ODatabaseRaw.java:246) ~[na:na]
... 31 common frames omitted
Wrapped by: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #22:24 (cluster: clusterB)
at com.orientechnologies.orient.core.db.raw.ODatabaseRaw.read(ODatabaseRaw.java:253) ~[na:na]
at com.orientechnologies.orient.core.db.record.ODatabaseRecordAbstract.executeReadRecord(ODatabaseRecordAbstract.java:1015) ~[na:na]
at com.orientechnologies.orient.core.tx.OTransactionOptimistic.loadRecord(OTransactionOptimistic.java:342) ~[na:na]
at com.orientechnologies.orient.core.db.record.ODatabaseRecordTx.load(ODatabaseRecordTx.java:261) ~[na:na]
at com.orientechnologies.orient.core.db.record.ODatabaseRecordTx.load(ODatabaseRecordTx.java:40) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.executeSearchRecord(OCommandExecutorSQLSelect.java:540) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.fetchFromTarget(OCommandExecutorSQLSelect.java:1014) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.fetchValuesFromIndexCursor(OCommandExecutorSQLSelect.java:1317) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searchForIndexes(OCommandExecutorSQLSelect.java:1233) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searchInClasses(OCommandExecutorSQLSelect.java:755) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbstract.assignTarget(OCommandExecutorSQLResultsetAbstract.java:134) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.assignTarget(OCommandExecutorSQLSelect.java:505) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.executeSearch(OCommandExecutorSQLSelect.java:489) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.execute(OCommandExecutorSQLSelect.java:457) ~[na:na]
at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:60) ~[na:na]
at com.orientechnologies.orient.core.storage.OStorageEmbedded.executeCommand(OStorageEmbedded.java:94) ~[na:na]
at com.orientechnologies.orient.core.storage.OStorageEmbedded.command(OStorageEmbedded.java:83) ~[na:na]
at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstract.java:59) ~[na:na]
at com.tinkerpop.blueprints.impls.orient.OrientGraphCommand.execute(OrientGraphCommand.java:28) ~[na:na]
...custom code...
at com.tinkerpop.frames.FramedElement.invoke(FramedElement.java:83) ~[na:na]
at com.sun.proxy.$Proxy51.getByName(Unknown Source) ~[na:na]
...custom code...
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0]
at java.lang.Thread.run(Thread.java:744) [na:1.8.0]
```

After the above errors, the following occurs when the data involved in the locks above are touched.

```
Transaction was rolled back more times than it was started.
com.orientechnologies.orient.core.exception.OTransactionException: Transaction was rolled back more times than it was started.
at com.orientechnologies.orient.core.tx.OTransactionOptimistic.rollback(OTransactionOptimistic.java:292) ~[na:na]
at com.orientechnologies.orient.core.db.record.ODatabaseRecordTx.commit(ODatabaseRecordTx.java:152) ~[na:na]
at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:504) ~[na:na]
at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:496) ~[na:na]
at com.tinkerpop.blueprints.impls.orient.OrientTransactionalGraph.commit(OrientTransactionalGraph.java:110) ~[na:na]
...custom code...
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0]
at java.lang.Thread.run(Thread.java:744) [na:1.8.0]
```

**Comment by azzazeli** [comment]
Hi.
I have the same issue reproduced on version 1.7.10

Here is my log:
```
2015-01-24 17:11:09 ERROR ONetworkProtocolBinary(OrientDB <- BinaryClient (/10.30.24.194:54607)) - Sent run-time exception to the client /10.30.24.194:54607: com.orientechnologies.common.concur.OTimeoutException: Timeout on acquiring exclusive lock against resource of class: class com.orientechnologies.common.concur.resource.OSharedResourceAdaptiveExternal with timeout=30000
Owner thread : Thread[OrientDB <- BinaryClient (/10.30.24.73:53810),5,OrientDB]
	at com.sun.jna.Native.free(Native Method)
	at com.orientechnologies.nio.OJNADirectMemory.free(OJNADirectMemory.java:44)
	at com.orientechnologies.common.directmemory.ODirectMemoryPointer.free(ODirectMemoryPointer.java:176)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog$LogSegment.readRecord(OWriteAheadLog.java:435)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog.read(OWriteAheadLog.java:918)
	at com.orientechnologies.orient.core.storage.impl.local.OStorageLocalAbstract.undoOperation(OStorageLocalAbstract.java:185)
	at com.orientechnologies.orient.core.storage.impl.local.OStorageLocalAbstract.rollbackStorageTx(OStorageLocalAbstract.java:149)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.rollback(OLocalPaginatedStorage.java:1152)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.commit(OLocalPaginatedStorage.java:1111)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit(OTransactionOptimistic.java:469)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(OTransactionOptimistic.java:148)
	at com.orientechnologies.orient.core.db.record.ODatabaseRecordTx.commit(ODatabaseRecordTx.java:142)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:511)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:503)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.commit(ONetworkProtocolBinary.java:1096)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:344)
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:170)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:45)

com.orientechnologies.common.concur.OTimeoutException: Timeout on acquiring exclusive lock against resource of class: class com.orientechnologies.common.concur.resource.OSharedResourceAdaptiveExternal with timeout=30000
Owner thread : Thread[OrientDB <- BinaryClient (/10.30.24.73:53810),5,OrientDB]
	at com.sun.jna.Native.free(Native Method)
	at com.orientechnologies.nio.OJNADirectMemory.free(OJNADirectMemory.java:44)
	at com.orientechnologies.common.directmemory.ODirectMemoryPointer.free(ODirectMemoryPointer.java:176)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog$LogSegment.readRecord(OWriteAheadLog.java:435)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog.read(OWriteAheadLog.java:918)
	at com.orientechnologies.orient.core.storage.impl.local.OStorageLocalAbstract.undoOperation(OStorageLocalAbstract.java:185)
	at com.orientechnologies.orient.core.storage.impl.local.OStorageLocalAbstract.rollbackStorageTx(OStorageLocalAbstract.java:149)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.rollback(OLocalPaginatedStorage.java:1152)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.commit(OLocalPaginatedStorage.java:1111)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit(OTransactionOptimistic.java:469)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(OTransactionOptimistic.java:148)
	at com.orientechnologies.orient.core.db.record.ODatabaseRecordTx.commit(ODatabaseRecordTx.java:142)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:511)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:503)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.commit(ONetworkProtocolBinary.java:1096)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:344)
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:170)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:45)

	at com.orientechnologies.common.concur.resource.OSharedResourceAdaptive.throwTimeoutException(OSharedResourceAdaptive.java:184)[orient-commons-1.7.10.jar:1.7.10]
	at com.orientechnologies.common.concur.resource.OSharedResourceAdaptive.acquireExclusiveLock(OSharedResourceAdaptive.java:127)[orient-commons-1.7.10.jar:1.7.10]
	at com.orientechnologies.common.concur.resource.OSharedResourceAdaptiveExternal.acquireExclusiveLock(OSharedResourceAdaptiveExternal.java:31)[orient-commons-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.doClose(OLocalPaginatedStorage.java:1920)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.close(OLocalPaginatedStorage.java:341)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.storage.OStorageAbstract.close(OStorageAbstract.java:106)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.db.raw.ODatabaseRaw.close(ODatabaseRaw.java:520)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.db.ODatabaseWrapperAbstract.close(ODatabaseWrapperAbstract.java:81)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.db.record.ODatabaseRecordAbstract.close(ODatabaseRecordAbstract.java:452)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.db.record.ODatabaseRecordTx.close(ODatabaseRecordTx.java:187)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.db.record.ODatabaseRecordAbstract.open(ODatabaseRecordAbstract.java:321)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.core.db.ODatabaseWrapperAbstract.open(ODatabaseWrapperAbstract.java:49)[orientdb-core-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.server.OServer.openDatabase(OServer.java:557)[orientdb-server-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.openDatabase(ONetworkProtocolBinary.java:630)[orientdb-server-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:229)[orientdb-server-1.7.10.jar:1.7.10]
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:170)[orientdb-server-1.7.10.jar:1.7.10]
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:45)[orient-commons-1.7.10.jar:1.7.10]
```

**Comment by lvca** [comment]
@azzazeli this issue has been fixed in 2.0M2, so please upgrade to 2.0
